ProjectSidewalk/SidewalkWebpage
# Add username restrictions and avoid raw html code in favor of directly replacing text content

I added username restrictions to prevent `"`, `'`, `<`, `>`, `&` from being characters in usernames. I also replaced some raw html that used dynamic, user-specified variables, in favor of setting the text content and attribute values directly.

This PR is meant to fix a security loophole. Mikey and I discussed this in Slack DMs and more info can be requested from either of us if interested :)

##### Testing instructions
1. Try to create an account with a bad username and verify that you're prevented from doing so
2. Test each feature that was modified in the code, where raw html was replaced with text/attribute content.

##### Things to check before submitting the PR 



NOTE: This is currently a draft because a lot of the changes I've made (except for the username restriction) are untested. I'm quite confident they'll work but further testing is needed to be 100% sure.

EDIT: It seems my code formatter made some changes that I didn't notice. I think they're fine but please don't hesitate to let me know if I should stick with the old formatting! I realize that it makes it kind of hard to see where I made changes to the code because of all the noise from my formatter's changes.
